android-sdk/README.md

```diff
@@ -0,0 +1,93 @@
+
+## Proposed solution
+
+### Core SDK
+
+The Core SDK will be the package that all service modules inherit from.  It will be responsible for managing network requests, logging, error handling and resolution, loading configuration for mobile-core services, and other cross cutting concerns.  Modules will interface with the core SDK by implementing the necessary interfaces provided and annotating the implementations so that the Core SDK will be able to correctly manage them as well.  The Core SDK will also allow some of its internals to be exposed through interfaces for purposes such as testing, bespoke integrations, and management of additional resources.
+
+The Core SDK will provide a request lifecycle that the Psuedocode section will demonstrate.
+
+### Service Modules
+
+Service modules will each represent a singular service that is supportd by mobile-core.  We will use, where available and practical, existing libraries for services and in those cases the modules will provide the necessary implementation of Core APIs to make them compatable with the Core SDK lifecycles and features.  The first Services we are looking at integrations will be Keycloak, fh-sync-service, and a RESTful API service module.
+
+### Use cases and psuedo code examples
+
+#### Defining a Service Module
+A service interface will be responsible for building requests, parsing responses, handling errors.  Services may depend on each other and the method of service composition is currently under evaluation. However, services will have a resolution order where a request is passed along a chain until it is ready to be executed.
+#### Create a Request
+A request is a object that is ready to be executed. The request contains in itself any authentication tokens, headers, etc necessary to connect to a service running in a mobile-core environment.
+For this example, let’s say the user is writing a photo sharing application.  They will probably in their application have an interface that looks like this : 
+```
+public interface UploadService {
+   public RequestHandle upload(Picture mypicture);
+}
+```
+There will be, in OpenShift, a Mobile application, a sync server, a keycloak server, and a storage server.  An implementation of this interface could load a keycloak token from the device, validate the token, start a new OkHttpRequest object, Upload the file etc etc.  What I am getting at is that this is a very laborious process.  However, let’s annotate this call a bit.
+```
+@MobileService(requires = {KeycloakService.class, SyncService.class, StorageService.class},
+                  appKey=”xyz”)
+public interface UploadService {
+   @ServiceMethod
+   public RequestHandle upload(Picture mypicture);
+}
+```
+In this world, the `@MobileService` annotation would signal to the core SDK that this interface should be implemented by our libraries and it requires a request to include the Keycloak, Sync, and Storage.
```

Comment:
Core SDK will scan application to check for @MobileService annotation?

Reply:
"Yes".

If we make a plugin, the scan can be done at compile time and we can generate the instanciation code then.  (Dagger 2 style).  Without the plugin we can scan at startup/lazy loading (Dagger classic style)